src/palindrom.js

```diff
@@ -632,17 +641,35 @@ var Palindrom = (function () {
   };
 
   Palindrom.prototype.observe = function () {
-    this.observer = this.jsonpatch.observe(this.obj, this.filterChangedCallback.bind(this));
+    /* if we haven't ever proxified our object,
+    this means it's the first observe call,
+    let's proxify it then! */
+    if (!this.isObjectProxified) {
+        this.obj = this.jsonPatcherProxy.observe(true, this.filterChangedCallback.bind(this));
+        this.isObjectProxified = true;
+    }
+    /* we are already observing, just enable event emitting. */
+    else {
+        this.jsonPatcherProxy.switchObserverOn();
```

Comment:
Legit point, albeit it's already covered.

Unlike `jsonpatch`, this `jsonPatcherProxy` is an instance of JSONPatcherProxy, not a class/module. So this `jsonPatcherProxy` represents only this object. If you want to obesrve two objects, you'll have to instantiate another instance as in:

```js
var jsonPatcherProxyForMyObj = new JSONPatcherProxy( myObj );
```
and
```js
var jsonPatcherProxyForYourObj = new JSONPatcherProxy( yourObj );
```

And IMHO, this makes it much better and faster. Because this moves garbage collection out of our way to the native VM. When you `dispose` an instance of `JSONPatcherProxy`, you won't need to care about the `obj` inside it, and patches history array, the proxy, and all the needed memory. As opposed to the traditional observers array where you have to remove and add instances in JS environment with self-written potentially buggy code and kind of garbage collect yourself. 

As always, I'm open to any suggestion you don't agree.